share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,628 @@
+## Wake Tutorial
+
+### Invoking wake
+
+1. Create a wake database: `wake --init .`
+
+2. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`
+
+* If you want to display a string you can that using the below function by writing the string to be displayed inisde "" and call the function abc . Ex: wake -x 'abc'
+```
+def abc = "Hello World"
+```
+
+* The other way to do this is `wake -x '"Hello World"'`
+
+### Defining Functions
+A function may be defines using the keyword def. Simple function definition takes the form
+`def <fun name> <parameter> = <expression>`
```

Comment:
```suggestion
`def <function name> <parameter list> = <expression>`
```

Can I specify a return value? how is that done

Reply:
can i specify types for the parameters?

Reply:
We cannot specify explicitly the return type and type of parameters. Usually return value is specified by the last line in the given function.